### PR TITLE
Update Harpy

### PR DESCRIPTION
**Pull request by pdimens**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by pdimens** [comment]
I don't understand, I have added the `run_exports:` directive to the recipe and the linter is still suggesting I am not:
I have followed the instructions [here](https://docs.conda.io/projects/conda-build/en/stable/resources/define-metadata.html#export-runtime-requirements) but to no avail:
```
ERROR: recipes/harpy/meta.yaml:0: missing_run_exports: Recipe should have a run_export statement that ensures correct pinning in downstream packages
```

@dpryan79

File: recipes/harpy/meta.yaml [hunk 1]
```diff
@@ -10,42 +10,52 @@ source:
   sha256: '{{ sha256 }}'
 
 build:
-  skip: True  # [osx]
   number: 0
+#  skip: True  # [osx]
+  noarch: python
+  script: {{ PYTHON }} -m pip install --no-deps .
```
Comment (mencian) [inline recipes/harpy/meta.yaml:16]:
```suggestion
  script: {{ PYTHON }} -m pip install --no-deps .
  run_exports:
    - {{ pin_subpackage('harpy', max_pin="x") }}
```

File: recipes/harpy/meta.yaml [hunk 2]
```diff
@@ -62,7 +72,7 @@ about:
 extra:
   container:
     extended-base: True
-  recipe-maintainers:
+  maintainers:
```
Comment (mencian) [inline recipes/harpy/meta.yaml:75]:
```suggestion
  recipe-maintainers:
```

**Review by mencian — COMMENTED** [review]
A couple of comments in-line, thanks!

**Review by pdimens — COMMENTED** [review]
apply changes

**Comment by pdimens** [comment]
@mencian there is one precompiled binary in this package that is preventing the testing from going through with the current build configs:
https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=39616&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=4dddc55b-10bc-50fc-ac68-7ff899b7a030&l=946
```
[noarch_python] Noarch package contains binary script: demuxGen1
```

**Comment by pdimens** [comment]
@mencian I believe I figured out the build part, but I'm getting confusing/conflicting dep solving errors during the test step:
```
The reported errors are:
- Encountered problems while solving:
-   - package htslib-1.17-h81da01d_2 requires libzlib >=1.2.13,<1.3.0a0, but none of the providers can be installed
```

The `meta.yml` file includes `libzlib` as a dependency, so I'm not sure why it is fussing.

**Comment by pdimens** [comment]
@BiocondaBot please add label

**Review by dpryan79 — APPROVED** [review]

**Pull request merged**